tests/PhotoMuPair_TEST.cxx

```diff
@@ -0,0 +1,139 @@
+#include "gtest/gtest.h"
+#include <nlohmann/json.hpp>
+
+#include "PROPOSAL/crosssection/Factories/PhotoMuPairProductionFactory.h"
+#include "PROPOSAL/secondaries/parametrization/photomupairproduction/PhotoMuPairProductionBurkhardtKelnerKokoulin.h"
+#include "PROPOSAL/medium/MediumFactory.h"
+#include "PROPOSAL/particle/ParticleDef.h"
+#include "PROPOSALTestUtilities/TestFilesHandling.h"
+
+using namespace PROPOSAL;
+
+ParticleDef getParticleDef(const std::string& name)
+{
+    if (name == "Gamma") {
+        return GammaDef();
+    } else {
+        EXPECT_TRUE(false);
+        return MuMinusDef();
+    }
+}
+
+TEST(PhotoMuPair, Test_of_dNdx)
+{
+    std::ifstream in;
+    getTestFile("PhotoMuPair_dNdx.txt", in);
+
+    std::string particleName;
+    std::string mediumName;
+    double multiplier;
+    double energy;
+    std::string parametrization;
+    double dNdx_stored;
+    double dNdx_new;
+
+    std::cout.precision(16);
+
+    while (in >> particleName >> mediumName >> multiplier >> energy
+        >> parametrization >> dNdx_stored) {
+
+        ParticleDef particle_def = getParticleDef(particleName);
+        auto medium = CreateMedium(mediumName);
+
+        nlohmann::json config;
+        config["parametrization"] = parametrization;
+
+        auto cross = make_photomupairproduction(particle_def, *medium, false, config);
+
+        dNdx_new = cross->CalculatedNdx(energy);
+        EXPECT_NEAR(dNdx_new, dNdx_stored, 1e-5 * dNdx_stored);
+
+        auto dEdx = cross->CalculatedEdx(energy);
+        EXPECT_EQ(dEdx, 0); // only-stochastic interaction
+
+        auto dE2dx = cross->CalculatedE2dx(energy);
+        EXPECT_EQ(dE2dx, 0); // only-stochastic interaction
+
+        auto v_loss = cross->CalculateStochasticLoss(0, energy, dNdx_new);
+        EXPECT_EQ(v_loss, 1); // fatal interaction
+    }
+}
+
+TEST(PhotoMuPair, Test_of_dNdx_Interpolant)
+{
+    std::ifstream in;
+    getTestFile("PhotoMuPair_dNdx.txt", in);
+
+    std::string particleName;
+    std::string mediumName;
+    double multiplier;
+    double energy;
+    std::string parametrization;
+    double dNdx_stored;
+    double dNdx_new;
+
+    while (in >> particleName >> mediumName >> multiplier >> energy
+        >> parametrization >> dNdx_stored) {
+
+        ParticleDef particle_def = getParticleDef(particleName);
+        auto medium = CreateMedium(mediumName);
+
+        nlohmann::json config;
+        config["parametrization"] = parametrization;
+
+        auto cross = make_photomupairproduction(particle_def, *medium, true, config);
+
+        dNdx_new = cross->CalculatedNdx(energy);
+        EXPECT_NEAR(dNdx_new, dNdx_stored, 1e-3 * dNdx_stored);
```

Comment:
Can the accuracy be increased to 1e-6 or is this the limit?

Reply:
No, the interpolation precision is actually limited with the current settings since the interpolation is not "trivial" for all energies.

For example for 1e3 to 1e4 MeV:

![image](https://user-images.githubusercontent.com/15159319/158162014-bb24dbae-6c33-4687-bfe0-91bc2430e1d8.png)